src/pydata_sphinx_theme/__init__.py

```diff
@@ -131,19 +131,39 @@ def generate_nav_html(kind, startdepth=None, show_nav_level=1, **kwargs):
             for ul in soup("ul", recursive=False):
                 ul.attrs["class"] = ul.attrs.get("class", []) + ["nav", "bd-sidenav"]
 
+            # restructing html to handle part collapsing
+            partcaptions = soup.find_all("p", attrs={"class": "caption"})
```

Comment:
Might want a more specific selective here to ensure we know this is within the sidebar, e.g. `soup.select("#bd-docs-nav p.caption")`

Reply:
@choldgraf can't use this selector as the soup literally starts from the `p` caption tag. So, it does not have `#bd-docs-nav`. So, had to use this simple selector as there is no parent for the caption in soup.

Reply:
ahh I see, if the soup starts that specifically then this is no big deal, I thought it was soup for the whole page